foreshadow/transformers/transformers.py

```diff
@@ -58,187 +92,288 @@ def wrap_transformer(transformer):
+                try:
+                    with warnings.catch_warnings(record=True) as w:
+                        value = getattr(self, key, None)
+                    if len(w) and w[0].category == DeprecationWarning:
+                        # if the parameter is deprecated, don't show it
+                        continue
+                finally:
+                    warnings.filters.pop(0)
+
+                # XXX: should we rather test if instance of estimator?
+                if deep and hasattr(value, "get_params"):
+                    deep_items = value.get_params().items()
+                    self_params.update(
+                        (key + "__" + k, val) for k, val in deep_items
+                    )
+                self_params[key] = value
+            return self_params
+
+        def fit(self, X, *args, **kwargs):
+            """Fit the estimtor or transformer, pandas enabled.
+
+            See transformer.
+
+            Args:
+                X: inputs
+                *args: arguments to transformer
+                **kwargs: keyword arguments to transformer
+
+            Returns:
+                self
+
+            """
+            df = check_df(X)
+
+            func = super(DFTransformer, self).fit
+            if df.empty and not isinstance(self, _Empty):
+                # this situation may happen when a transformer comes after the
+                # Empty transformer in a pipeline. Sklearn transformers will
+                # break on empty input and so we reroute to _Empty.
+                func = partial(_Empty.fit, self)
+            out = func(df, *args, **kwargs)
+            return out
 
+        def transform(self, X, y=None, *args, **kwargs):
+            """Transform inputs using fitted transformer. Pandas enabled.
+
+            See transformer
+
+            Args:
+                X: inputs
+                y: labels
+                *args: arguments to transformer
+                **kwargs: keyword arguments to transformer
+
+            Returns:
+                transformed inputs
+
+            Raises:
+                ValueError: if not a valid output type from transformer
+
+            """
+            df = check_df(X)
+
+            init_cols = [str(col) for col in df]
+            func = super(DFTransformer, self).transform
+            if df.empty and not isinstance(self, _Empty):
+                # this situation may happen when a transformer comes after the
+                # Empty transformer in a pipeline. Sklearn transformers will
+                # break on empty input and so we reroute to _Empty.
+                func = partial(_Empty.transform, self)
+
+            out = func(df, *args, **kwargs)
+
+            # determine name of new columns
+            name = self.name if self.name else type(self).__name__
+            out_is_transformer = hasattr(out, "__class__") and is_transformer(
+                out.__class__
+            )  # noqa: E127
```

Comment:
were black and flake8 conflicting here, is that why the noqa exists if so we should throw up and issue.

Reply:
Not sure, I tried getting the formatting to work but couldn't. I'll take another look